### ISSUE TITLE
Matching rule on array size gets inherited by all child arrays

### ISSUE THREAD

**Issue by artamonovkirill** [pact-foundation/pact-jvm#396]
Following pact:
```Groovy:
def provider = new PactBuilder()
provider {
    serviceConsumer 'Consumer'
    hasPactWith 'Provider'
    port 9000

    uponReceiving('request')
    withAttributes(
        method: 'get',
        path: PATH)
    willRespondWith(
        status: 200)
    withBody {
        parent minLike(2, 2) {
            child(['a'])
        }
    }
}
```
will generate a matching rule like:
```JSON
{
	"body": {
		"$.parent": {
			"matchers": [{
				"match": "type",
				"min": 2
			}]
		}
	}
}
```
that will fail on a provider response like:
```JSON
{
	"parent": [{
		"child": ["a"]
	}, {
		"child": ["a"]
	}]
}
```
with
```
java.lang.AssertionError: 
0 - $.parent.0.child -> {mismatch=Expected List(a) to have minimum 2, diff=}
```

Consumer & provider test code: https://github.com/artamonovkirill/pact-issues-examples/tree/master/src/test/groovy/com/github/artamonovkirill/pact/minlike

**Comment by uglyog** [comment]
This has been fixed in 4.1.25

**Comment by artamonovkirill** [comment]
Thanks 🙌
